core/src/plan/validate.rs

```diff
@@ -5,25 +5,34 @@ use {
         data::Schema,
         result::Result,
     },
-    std::{collections::HashMap, rc::Rc},
+    std::{collections::HashMap, ops::Deref, rc::Rc},
 };
 
 type SchemaMap = HashMap<String, Schema>;
 /// Validate user select column should not be ambiguous
 pub fn validate(schema_map: &SchemaMap, statement: &Statement) -> Result<()> {
-    if let Statement::Query(Query {
-        body: SetExpr::Select(select),
-        ..
-    }) = &statement
-    {
-        for select_item in &select.projection {
-            if let SelectItem::Expr {
-                expr: Expr::Identifier(ident),
-                ..
-            } = select_item
-            {
-                if let Some(context) = contextualize_stmt(schema_map, statement) {
-                    context.validate_duplicated(ident)?;
+    let query = match statement {
+        Statement::Query(query) => Some(query),
+        Statement::Insert { source, .. } => Some(source),
+        Statement::CreateTable { source, .. } => source.as_ref().map(Deref::deref),
```

Comment:
nit. without using `std::ops::Deref` you can simply use `source.as_deref()` here